yarn-project/foundation/src/promise/running-promise.ts

```diff
@@ -23,7 +28,11 @@ export class RunningPromise {
     const poll = async () => {
       while (this.running) {
         const hasRequested = this.requested !== undefined;
-        await this.fn();
+        try {
+          await this.fn();
+        } catch (err) {
```

Comment:
so no cases where we'd actually want the running promise to fail on error?

Reply:
Problem is a failure is an uncaught rejected promise, which suck.

Reply:
Just double checked, and by default it'll turn into an uncaught exception, which I understand would kill the process: https://nodejs.org/api/cli.html#--unhandled-rejectionsmode